docs/src/unbound_args.md

```diff
@@ -0,0 +1,42 @@
+# Unbound Type Parameters
+
+An unbound type parameter is a type parameter with a `where`,
+that does not occur in the signature of some dispatch of the method.
+
+## Examples
+
+The following methods each have `T` as an unbound type parameter:
+
+```@repl
+f(x::Int) where {T} = do_something(x)
+g(x::T...) where {T} = println(T)
+```
+
+In the cases of `f` above, the unbound type parameter `T` is neither
+present in the signature of the methods nor as a bound of another type parameter.
+Here, the type parameter `T` can be removed without changing any semantics.
+
+For signatures with `Vararg` (cf. `g` above), the type parameter is unbound for the 
+zero-argument case (e.g. `g()`).
+
+```@repl
+g(1.0, 2.0)
+g(1)
+g()
+```
+
+A possible fix would be to replace `g` by two methods.
+
+```@repl
```

Comment:
Currently, the documentation shows `g not defined`.
![image](https://github.com/JuliaTesting/Aqua.jl/assets/7488140/efff885c-aee9-467a-990d-1013b0be1980)

````suggestion
```@repl unbound
f(x::Int) where {T} = do_something(x)
g(x::T...) where {T} = println(T)
```

In the cases of `f` above, the unbound type parameter `T` is neither
present in the signature of the methods nor as a bound of another type parameter.
Here, the type parameter `T` can be removed without changing any semantics.

For signatures with `Vararg` (cf. `g` above), the type parameter is unbound for the 
zero-argument case (e.g. `g()`).

```@repl unbound
g(1.0, 2.0)
g(1)
g()
```

A possible fix would be to replace `g` by two methods.

```@repl unbound
````

After the fix, the function `g` has three methods, and the first method will never be called.
Is this okay?
This is why I suggest using `h` in [the suggestion](https://github.com/JuliaTesting/Aqua.jl/pull/250#discussion_r1408706061).

Reply:
````suggestion
```@repl unbound
f(x::Int) where {T} = do_something(x)
g(x::T...) where {T} = println(T)
```

In the cases of `f` above, the unbound type parameter `T` is neither
present in the signature of the methods nor as a bound of another type parameter.
Here, the type parameter `T` can be removed without changing any semantics.

For signatures with `Vararg` (cf. `g` above), the type parameter is unbound for the 
zero-argument case (e.g. `g()`).

```@repl unbound
g(1.0, 2.0)
g(1)
g()
```

A possible fix would be to replace `g` by two methods.

```@repl unbound2
````

That's true. I would like to keep the name the same though (as it states to `replace`). So I would make a different repl env instead.